spec/lib/miq_automation_engine/service_methods/miq_ae_service_spec.rb

```diff
@@ -46,6 +46,89 @@ def assert_readonly_instance(automate_method_script)
+      before { NotificationType.seed }
+
+      before(:each) do
+        @vm = FactoryGirl.create(:vm, :tenant => @root_tenant)
+      end
+
+      let(:ws) do
+        MiqAeEngine.instantiate("/EVM/AUTOMATE/test1?Vm::vm=#{@vm.id}", @owner)
+      end
+
+      def type_method_script
+        "vm = $evm.root['vm']
+        $evm.create_notification(:type => :vm_provisioned, :subject => vm, :message => 'this is a test')"
+      end
+
+      def invalid_type_method_script
+        "vm = $evm.root['vm']
+        $evm.create_notification(:type => :invalid_type, :subject => vm)"
+      end
+
+      def invalid_subject_method_script
+        "$evm.create_notification(:type => :vm_powered_off, :subject => nil)"
+      end
+
+      it "existing notification type, valid subject" do
+        @ae_method.update_attributes(:data => type_method_script)
+        ws.root(@ae_result_key)
+        notification = Notification.last
+        expect(notification.options[:message]).to eql("this is a test")
+        expect(notification).to have_attributes(
+          :notification_type_id => NotificationType.find_by_name(:vm_provisioned).id,
+          :subject_id           => @vm.id
+        )
+      end
+
+      it "invalid notification type " do
+        @ae_method.update_attributes(:data => invalid_type_method_script)
+        expect { ws.root }.to raise_error(MiqAeException::UnknownMethodRc)
+      end
+
+      it "invalid subject " do
+        @ae_method.update_attributes(:data => invalid_subject_method_script)
+        expect { ws.root }.to raise_error(MiqAeException::UnknownMethodRc)
+      end
+
+      it "default type of automate_user_info" do
+        method = "$evm.create_notification(:message => 'mary had a little lamb')"
+        @ae_method.update_attributes(:data => method)
+        invoke_ae
+        notification = Notification.last
+        expect(notification.options[:message]).to eql("mary had a little lamb")
+        expect(notification).to have_attributes(
+          :notification_type_id => NotificationType.find_by_name(:automate_user_info).id,
+          :subject_id           => @owner.id
+        )
+      end
+
+      it "type automate_user_warning" do
+        method = "$evm.create_notification(:level => 'warning', :message => 'test for automate user warning')"
+        @ae_method.update_attributes(:data => method)
+        invoke_ae
+        notification = Notification.last
+        expect(notification.options[:message]).to eql("test for automate user warning")
+        expect(notification).to have_attributes(
+          :notification_type_id => NotificationType.find_by_name(:automate_user_warning).id,
+          :subject_id           => @owner.id
+        )
+      end
+
+      it "type automate_user_success" do
+        method = "$evm.create_notification(:level => 'success', :audience => 'user', :message => 'testing123')"
+        @ae_method.update_attributes(:data => method)
+        invoke_ae
+        notification = Notification.last
```

Comment:
@tinaafitz 
The last could lead to sporadic failures, can we just rely on the create_notification returning a true/false.